test/runtests.jl

```diff
@@ -1,136 +1,136 @@
-            grid1,
-            round(Int, 0.3 * count[1]),
-            round(Int, 0.05 * count[2]),
-            round(Int, 0.5 * count[2])
-        )
-        @test (grid == grid1)
+        @testset "Trans lines" begin
+            grid = baseline_grid()
+            @test grid.trans_lines[1].capacity == 4900
+            @test length(trans_lines(grid)) == 179
+            @test test_connectivity(grid.bus_conn, false)
+            @test total_links(grid.bus_conn) == 179
+            @test mean_node_deg(grid.bus_conn) ≈ 2.435374149659
+            @test mean_shortest_path(adjacency(grid)) ≈ 7.693646165949373
+            @test mean_shortest_path(adjacency(grid), distance(buses(grid))) ≈ 133.34235584030938
+            @test robustness_line(grid.bus_conn, 10) > 1
+            @test robustness_node(grid.bus_conn, 10) > 1
+        end
+
+        @testset "Add substation" begin
+            grid = Grid(true)
+            @test size(grid.bus_conn) == (137, 137)
+            @test size(grid.sub_conn) == (43, 43)
+
+            add_gen!(grid, (11., 11.), 100, [11], ["ki"], reconnect = false)
+            add_load!(grid, (19.,12.), 100, 100, 100, reconnect = true)
+            add_substation!(
+                grid,
+                (11., 18.),
+                [11],
+                1000,
+                6000,
+                500,
+                Set(),
+                [grid.buses[139], grid.buses[138]]
+            )
+            @test size(grid.bus_conn) == (139, 139)
+            @test size(grid.sub_conn) == (44, 44)
+        end
+
+        @testset "Clustering grids" begin
+            grid = baseline_grid()
+            count = SyntheticGrids.count_bus_type(grid)
+            cluster!(
+                grid,
+                round(Int, 0.3 * count[1]),
+                round(Int, 0.05 * count[2]),
+                round(Int, 0.5 * count[2])
+            )
+            @test length(substations(grid)) == 47
+            @test grid.substations[1].population == 6193
+            @test grid.substations[47].generation == 16.1
+            @test test_connectivity(grid.sub_conn, false)
+            @test total_links(grid.sub_conn) == 70
+            @test mean_shortest_path(sub_connectivity(grid) .> 0) ≈ 4.1620642824807605
+            @test mean_shortest_path(
+                (sub_connectivity(grid) .> 0),
+                distance(substations(grid))
+                ) ≈ 111.6094877525516
+            @test robustness_line(grid.sub_conn, 10) > 1
+            @test robustness_node(grid.sub_conn, 10) > 1
+        end
     end
 
     @testset "Input and Output" begin
         @testset "Pandapower interface" begin
+            grid = baseline_grid()
             pgrid = to_pandapower(grid, DUMMYPPC)
-            @test length(pgrid[:trafo]) == 36
+            @test length(pgrid.trafo) == 31
             pgrid = SyntheticGrids.load_pp_grid(DUMMYPPC)
-            @test length(pgrid[:trafo]) == 36
+            @test length(pgrid.trafo) == 31
```

Comment:
This has to be looked into, as the new test is finding 5 fewer transformers from the data.

Reply:
Posting for posterity here, will also update the description and post sample code there.

Yesterday going through we were able to find that generating random numbers has differed between Julia0.6 and Julia1.0. This results in the adjacency matrices being different in versions and then changing the amount of transformers.